Unity-Technologies/ScriptableRenderPipeline
# HDRP - Fix decal issue in player

### Purpose of this PR
This PR fix issue when building a player with Decal. The stripper is removing shader decal variant causing a change of pass order which make decal fail at rutime. This PR change the hard coded number to be based on pass name instead to not fail.

In addition this PR allow to create emissive only decal (i.e we remove other variants and don't render non emissive part of the decal).

This Pr also include a shader warning fix + add an option to clear GBuffer (false by default) on frame settings

---
### Release Notes
- Fixed issue with Decal not rendering correctly in the player due to incorrect shader decal variant selection.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ X] Opened test project + Run graphic tests locally
- [ X] Built a player
- [ X] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=test%2Fdecal&automation-tools_branch=master&unity_branch=2019.2%2Fstaging&sort=1-asc
(launched with revisoin f7ca70490e38)

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?

**Halo Effect**: None, Low, Medium, High?

---
### Comments to reviewers
Notes for the reviewers you have assigned.
